llama-index-integrations/readers/llama-index-readers-s3/llama_index/readers/s3/base.py

```diff
@@ -7,14 +7,16 @@
 
 import warnings
 from typing import Callable, Dict, List, Optional, Union
+from datetime import datetime, timezone
+from pathlib import Path
 
-from llama_index.core.readers import SimpleDirectoryReader
+from llama_index.core.readers import SimpleDirectoryReader, BaseFilesystemReader
 from llama_index.core.readers.base import BaseReader, BasePydanticReader
 from llama_index.core.schema import Document
 from llama_index.core.bridge.pydantic import Field
 
 
-class S3Reader(BasePydanticReader):
+class S3Reader(BasePydanticReader, BaseFilesystemReader):
```

Comment:
I almost wonder if BaseFileSystemReader should just be subclassing BasePydanticReader? If we follow the dependency list of classes, now S3Reader inherits both BaseReader and BasePydanticReader lol

(This is where all the subclasses start to get confusing haha)

Reply:
agreed, this multiple inheritance is confusing.